### DISCUSSION TITLE
Multiple GPU with Fooocus

### DISCUSSION THREAD

**Discussion by adolvex** [lllyasviel/Fooocus#1262] [Q&A]
Hi everybody,

I was wondering, is it possible to use multiple GPU with Fooocus? I currently own a RTX 3070 8gb but I'm considering adding another 3070 if the multiple gpu could be possible.

And if I may ask, would be better to have a 3090 or dual 3070? Just for the practical future-selling purposes it would make more sense to me to have two 3070's, but if the results are that much more faster with a 3090 I might change it.

Thank you in advance for your answers!

**Comment by JamesC999** [comment]
The multi-GPU part of your post looks like it's somewhat answered in #292. The response is "parallel support has not (yet) been implemented." I don't know what the potential timeline would look like on implementing that, and I'm sure they are busy, but I'd like to know (assuming they have an estimate). This would be a game changer to me, and make this awesome project even better.

It's difficult for me to understand what that would look like, since I think parallel support would mean having two sets of images processing at the same time, in which case I wonder how they would approach preview loading.

As far as future-selling purposes, I think it's really difficult to predict what will be in demand. I take the approach of buy what I want, and deal with the marketplace once I'm ready to sell. I can't really provide good advice to that.

**Reply by mashb1t** [reply]
https://github.com/lllyasviel/Fooocus/pull/981 (see https://github.com/lllyasviel/Fooocus/pull/981#issuecomment-1904766195) is still awaiting review, but might be merged rather sooner than later to build the basis for multi GPU support.

**Reply by toutjavascript** [reply]
It would be GREAT !

**Reply by adolvex** [reply]
I'm also waiting for this feature to be released, It's going to change lots of things, thank you in advance for your hard work!

**Reply by JibJiggler** [reply]
It looks like it renders on GPU0.  I use GPU1 for my monitor.  Typically I use this for my Oculus, but I stumbled across this thread when I was playing with Fooocus for the first time. Yes, I know they are old. but as you
![Fooocus GPU usage](https://github.com/user-attachments/assets/89f984ee-5843-49e1-9921-cc84ac9f9b2a)
 can see GPU0 is doing more of the work.

**Reply by JibJiggler** [reply]
It did start like it was going to use GPU1.  Just give it a try.

**Reply by JibJiggler** [reply]
Next steps, I'll mess with the Batch file.

**Reply by mashb1t** [reply]
@JibJiggler multiple GPUs are currently not supported, you  have to start Fooocus twice and select different GPUs for each. Keep in mind that all ghe ressources are then used twice (by design), incl. RAM, VRAM, disk, etc.
Also see https://github.com/lllyasviel/Fooocus/discussions/2292
